.github/workflows/pull_request.yml

```diff
@@ -11,5 +11,66 @@ jobs:
   quality_checks:
     uses: ./.github/workflows/quality_checks.yml
 
+  get_issue_number:
+    runs-on: ubuntu-latest
+    needs: quality_checks
+    outputs:
+        issue_number: ${{steps.get_issue_number.outputs.result}}
+    steps:
+      - uses: actions/github-script@v6
+        name: get issue number
+        id: get_issue_number
+        with:
+          script: |
+            if (context.issue.number) {
+              // Return issue number if present
+              return context.issue.number;
+            } else {
+              // Otherwise return issue number from commit
+              return (
+                await github.rest.repos.listPullRequestsAssociatedWithCommit({
+                  commit_sha: context.sha,
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                })
+              ).data[0].number;
+            }
+          result-encoding: string
+  
+  get_commit_id:
+    runs-on: ubuntu-latest
+    outputs:
+      commit_id: ${{steps.commit_id.outputs.commit_id}}
+    steps:
+      - name: Get Commit ID
+        id: commit_id
+        run: echo "commit_id=${{github.sha}}" >> "$GITHUB_OUTPUT"
 
- 
+  list_ci_resources_cf_changes:
+    needs: [get_issue_number, get_commit_id]
+    uses: ./.github/workflows/cloudformation.yml
+    with:
+      target_environment: dev
+      stack_name: ci-resources
+      change_set_name: "ci-resources-${{github.event.number}}"
+      template: ./cloudformation/ci_resources.yml
+      capabilities: CAPABILITY_IAM
+      parameter_overrides: ParameterKey="SubjectClaimFilters",ParameterValue="'repo:NHSDigital/prescriptionsforpatients:*,repo:NHSDigital/electronic-prescription-service-account-resources:*,repo:NHSDigital/electronic-prescription-service-clinical-prescription-tracker:*,repo:NHSDigital/electronic-prescription-service-release-notes:*,repo:NHSDigital/prescriptions-for-patients:*,repo:NHSDigital/electronic-prescription-service-api:*'"
```

Comment:
do you think there is a way to have SubjectClaimFilters for each environment in a config file for readability and maintainability? All environments could be in one file, or could have one file per environment